internal/status/gateway.go

```diff
@@ -40,24 +40,32 @@ func UpdateGatewayStatusProgrammedCondition(gw *gwapiv1b1.Gateway, svc *corev1.S
 			}
 		}
 
-		var gwAddrs []gwapiv1b1.GatewayAddress
+		if svc.Spec.Type == corev1.ServiceTypeClusterIP && len(svc.Spec.ClusterIPs) != 0 {
```

Comment:
nit: len check not needed :)

Reply:
can you add a similar check on L29 (svc.Spec.Type == Loadbalancer) ?

Reply:
did it